db/record.c

```diff
@@ -2869,6 +2888,19 @@ static int check_blob_buffers(struct ireq *iq, blob_buffer_t *blobs,
     return 0;
 }
 
+static int check_blob_sizes(struct ireq *iq, blob_buffer_t *blobs, int maxblobs)
+{
+    for (int i = 0; i < maxblobs; i++) {
+        if (blobs[i].exists && blobs[i].length > MAXBLOBLENGTH) {
```

Comment:
`blobs[i].length` is `size_t`. We set blob length to `-2` if the blob is not used by the query. Therefore you need to cast `blobs[i].length` to signed.